telethon/events/callbackquery.py

```diff
@@ -24,14 +24,20 @@ class CallbackQuery(EventBuilder):
             A UTF-8 string can also be given, a regex or a callable. For
             instance, to check against ``'data_1'`` and ``'data_2'`` you
             can use ``re.compile(b'data_')``.
+
+        pattern (`str`, `callable`, `Pattern`, optional):
+            If set, only buttons with payload matching this pattern will be handled.
+            You can specify a regex-like string which will be matched
+            against the message, a callable function that returns ``True``
```

Comment:
First line:
> only buttons with payload matching this pattern 

Second-third line:
> will be matched against the message

So... does it match button payload (data) or message contents?